### PR TITLE
add resource command

### PR DESCRIPTION
**Pull request by zhuangzhuang131419**
## Why are these changes needed?

Using container.resource.cpu.limit as num-cpus default value in rayStartParams, it will help to reduce configuration redundancy. When we want to change the container resource config, we don't need to update rayStartParams.

It's the same with memory,gpu.

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Jeffwan** [comment]
Address https://github.com/ray-project/kuberay/issues/152

File: ray-operator/config/samples/ray-cluster.mini.yaml [hunk 1]
```diff
@@ -53,6 +53,15 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: status.podIP
+          resources:
+            limits:
+              cpu: 2
+              memory: 4Gi
+              gpu: 1
```
Comment (Jeffwan) [inline ray-operator/config/samples/ray-cluster.mini.yaml:60]: I think mini is being used by many users to verify the control plane functionality. I would suggest to use less resources like 1c 2gi and remove gpu. WDYT?

**Review by Jeffwan — COMMENTED** [review]
rest part LGTM

File: ray-operator/controllers/common/pod.go [hunk 2]
```diff
@@ -260,7 +260,31 @@ func setMissingRayStartParams(rayStartParams map[string]string, nodeType rayiov1
...
+func concatenateContainerCommand(nodeType rayiov1alpha1.RayNodeType, rayStartParams map[string]string, resource v1.ResourceRequirements) (fullCmd string) {
+
+	if _, ok := rayStartParams["num-cpus"]; !ok {
+		cpu := resource.Limits[v1.ResourceCPU]
+		if !cpu.IsZero() {
+			rayStartParams["num-cpus"] = strconv.FormatInt(cpu.Value(), 10)
+		}
+	}
+
+	if _, ok := rayStartParams["object-store-memory"]; !ok {
```
Comment (chenk008) [inline ray-operator/controllers/common/pod.go:272]: `object-store-memory` is not equals to container memory.  Maybe `--memory` will be a matching option.

File: ray-operator/controllers/common/pod.go [hunk 3]
```diff
@@ -260,7 +260,31 @@ func setMissingRayStartParams(rayStartParams map[string]string, nodeType rayiov1
...
+	}
+
+	if _, ok := rayStartParams["num-gpus"]; !ok {
+		gpu := resource.Limits["gpu"]
+		if !gpu.IsZero() {
+			rayStartParams["num-gpus"] = strconv.FormatInt(gpu.Value(), 10)
+		}
+	}
+
+	log.Info("concatenate container command", "ray start params", rayStartParams)
```
Comment (chenk008) [inline ray-operator/controllers/common/pod.go:286]: We might use debug log here.

File: ray-operator/config/samples/ray-cluster.mini.yaml (hunk 1 above)
Comment (zhuangzhuang131419) [reply ray-operator/config/samples/ray-cluster.mini.yaml:60]: fix it

File: ray-operator/controllers/common/pod.go (hunk 2 above)
Comment (zhuangzhuang131419) [reply ray-operator/controllers/common/pod.go]: fix it

File: ray-operator/config/samples/ray-cluster.mini.yaml (hunk 1 above)
Comment (chenk008) [reply ray-operator/config/samples/ray-cluster.mini.yaml:60]: gpu is still there.

**Comment by Jeffwan** [comment]
@chenk008 Can you help double check and approve the change?

**Review by chenk008 — APPROVED** [review]
LGTM

**Pull request merged**